### ISSUE TITLE
Schema markup only shows sale price when product is on sale

### ISSUE THREAD

**Issue by galbaras** [woocommerce/woocommerce#24662]
**Describe the bug**
The schema markup for products on sale only includes the sale price, defaulting to the end of the following year for the duration of the sale.

From what I understand, the markup should include 2 offers: one with the "priceValidUntil" field and sale price, and the other with the regular price.

**To Reproduce**
1. Edit a product
2. Set a sale price
3. Save
4. Display the product
5. View page source code

**Expected behavior**
Offers for both regular and sale price, with the priceValidUntil property on the sale offer.

**Isolating the problem (mark completed items with an [x]):**
- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**WordPress Environment**
<details>
```
### WordPress Environment ###

WordPress address (URL): https://dev.get-business-online.com/pacificsprings
Site address (URL): https://dev.get-business-online.com/pacificsprings
WC Version: 3.7.0
REST API Version: ✔ 1.0.2
Log Directory Writable: ✔
WP Version: 5.2.3
WP Multisite: –
WP Memory Limit: ❌ 51 MB - We recommend setting memory to at least 64MB. See: Increasing memory allocated to PHP
WP Debug Mode: –
WP Cron: –
Language: en_AU
External object cache: –

### Server Environment ###

Server Info: LiteSpeed
PHP Version: 7.3.9
PHP Post Max Size: 64 MB
PHP Time Limit: 0
PHP Max Input Vars: 3000
cURL Version: 7.62.0
OpenSSL/1.0.2k

SUHOSIN Installed: –
MySQL Version: 10.2.27-MariaDB
Max Upload Size: 64 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 3.6.5
WC Database Prefix: wp_ps_
MaxMind GeoIP Database: ✔
Total Database Size: 91.40MB
Database Data Size: 61.97MB
Database Index Size: 29.43MB
wp_ps_woocommerce_sessions: Data: 0.09MB + Index: 0.02MB + Engine InnoDB
wp_ps_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_ps_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_ps_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_ps_woocommerce_order_items: Data: 0.16MB + Index: 0.06MB + Engine InnoDB
wp_ps_woocommerce_order_itemmeta: Data: 1.52MB + Index: 0.69MB + Engine InnoDB
wp_ps_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_ps_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_ps_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_ps_woocommerce_shipping_zone_locations: Data: 0.05MB + Index: 0.03MB + Engine InnoDB
wp_ps_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_ps_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_ps_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_ps_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
iwp_clone_stats: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
IWP_processed_iterator: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_ps_ac_areas: Data: 0.01MB + Index: 0.02MB + Engine MyISAM
wp_ps_apto_sort_list: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_ps_cartflows_ca_cart_abandonment: Data: 0.00MB + Index: 0.01MB + Engine MyISAM
wp_ps_cartflows_ca_email_history: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_ps_cartflows_ca_email_templates: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_ps_cartflows_ca_email_templates_meta: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_ps_cf7dbplugin_st: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_ps_cf7dbplugin_submits: Data: 0.47MB + Index: 0.31MB + Engine InnoDB
wp_ps_commentmeta: Data: 0.03MB + Index: 0.03MB + Engine MyISAM
wp_ps_comments: Data: 0.17MB + Index: 0.08MB + Engine MyISAM
wp_ps_ewwwio_images: Data: 0.50MB + Index: 0.21MB + Engine MyISAM
wp_ps_ewwwio_queue: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_ps_feedmanager_channel: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_ps_feedmanager_country: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_ps_feedmanager_feed_status: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_ps_feedmanager_field_categories: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_ps_feedmanager_product_feed: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_ps_feedmanager_product_feedmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_ps_feedmanager_source: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_ps_itsec_distributed_storage: Data: 3.63MB + Index: 0.00MB + Engine MyISAM
wp_ps_itsec_fingerprints: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_ps_itsec_geolocation_cache: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_ps_itsec_lockouts: Data: 0.02MB + Index: 0.08MB + Engine InnoDB
wp_ps_itsec_log: Data: 4.52MB + Index: 0.20MB + Engine InnoDB
wp_ps_itsec_logs: Data: 0.58MB + Index: 0.04MB + Engine MyISAM
wp_ps_itsec_temp: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_ps_iwp_backup_status: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_ps_iwp_file_list: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_ps_iwp_processed_iterator: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_ps_links: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_ps_options: Data: 0.97MB + Index: 0.06MB + Engine MyISAM
wp_ps_postmeta: Data: 4.55MB + Index: 1.77MB + Engine MyISAM
wp_ps_posts: Data: 1.18MB + Index: 0.29MB + Engine MyISAM
wp_ps_redirection_404: Data: 41.56MB + Index: 23.16MB + Engine InnoDB
wp_ps_redirection_groups: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_ps_redirection_items: Data: 0.03MB + Index: 0.04MB + Engine MyISAM
wp_ps_redirection_logs: Data: 0.03MB + Index: 0.02MB + Engine MyISAM
wp_ps_termmeta: Data: 0.00MB + Index: 0.01MB + Engine MyISAM
wp_ps_terms: Data: 0.01MB + Index: 0.02MB + Engine MyISAM
wp_ps_term_relationships: Data: 0.02MB + Index: 0.04MB + Engine MyISAM
wp_ps_term_sorting: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_ps_term_taxonomy: Data: 0.01MB + Index: 0.01MB + Engine MyISAM
wp_ps_usermeta: Data: 0.02MB + Index: 0.02MB + Engine MyISAM
wp_ps_users: Data: 0.00MB + Index: 0.01MB + Engine MyISAM
wp_ps_vslider: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_ps_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_ps_wc_product_meta_lookup: Data: 0.00MB + Index: 0.01MB + Engine MyISAM
wp_ps_wc_tax_rate_classes: Data: 0.00MB + Index: 0.01MB + Engine MyISAM
wp_ps_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_ps_wpsc_also_bought: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_ps_wpsc_cart_contents: Data: 0.11MB + Index: 0.02MB + Engine MyISAM
wp_ps_wpsc_cart_item_meta: Data: 0.08MB + Index: 0.20MB + Engine InnoDB
wp_ps_wpsc_checkout_forms: Data: 0.00MB + Index: 0.01MB + Engine MyISAM
wp_ps_wpsc_claimed_stock: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_ps_wpsc_coupon_codes: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_ps_wpsc_currency_list: Data: 0.02MB + Index: 0.01MB + Engine MyISAM
wp_ps_wpsc_download_status: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_ps_wpsc_meta: Data: 0.04MB + Index: 0.05MB + Engine MyISAM
wp_ps_wpsc_product_rating: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_ps_wpsc_purchase_logs: Data: 0.11MB + Index: 0.04MB + Engine MyISAM
wp_ps_wpsc_purchase_meta: Data: 0.17MB + Index: 0.41MB + Engine InnoDB
wp_ps_wpsc_region_tax: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_ps_wpsc_submited_form_data: Data: 0.57MB + Index: 0.54MB + Engine MyISAM
wp_ps_wpsc_visitors: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_ps_wpsc_visitor_meta: Data: 0.17MB + Index: 0.11MB + Engine InnoDB
wp_ps_yoast_seo_links: Data: 0.01MB + Index: 0.01MB + Engine MyISAM
wp_ps_yoast_seo_meta: Data: 0.00MB + Index: 0.01MB + Engine MyISAM

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (30) ###

InfiniteWP - Client: by Revmakx – 1.9.4.2
AddToAny Share Buttons: by AddToAny – 1.7.37
AJAX Search for WooCommerce: by Damian Góra – 1.5.0
iThemes Security: by iThemes – 7.4.1
Classic Editor: by WordPress Contributors – 1.5
Honeypot for Contact Form 7: by Nocean – 1.14.1
Contact Form 7: by Takayuki Miyoshi – 5.1.4
EWWW Image Optimizer: by Exactly WWW – 4.9.3
Flamingo: by Takayuki Miyoshi – 2.1
Custom Extensions and Styles: by Gal Baras – 2.14
jQuery Responsive Select Menu: by MIGHTYminnow – 1.5.0
Media File Renamer (Auto Rename): by Jordy Meow – 4.5.7
Redirection: by John Godley – 4.4
Regenerate Thumbnails: by Alex Mills (Viper007Bond) – 3.1.1
Special Recent Posts FREE Edition: by Luca Grandicelli – 2.0.4
Video Thumbnails: by Sutherland Boswell – 2.12.3
WooCommerce Cart Abandonment Recovery: by CartFlows Inc – 1.1.9 – Not tested with the active version of WooCommerce
WooCommerce Checkout Optimization by Cartimize: by Cartimize – 1.0.2
Woocommerce Minimum and Maximum Quantity: by Ashok G – 2.0.8 – Not tested with the active version of WooCommerce
WooCommerce Minicart: by Ahmad Shyk – 1.0.4 – Not tested with the active version of WooCommerce
Preview E-mails for WooCommerce: by Digamber Pradhan – 1.5.1 – Not tested with the active version of WooCommerce
WooCommerce Buy One Get One Free: by Oscar Gare – 1.3.5
WooCommerce Force Sells: by WooCommerce – 1.1.20
WooCommerce eWAY Payment Gateway: by WooCommerce – 3.1.22
WooCommerce: by Automattic – 3.7.0
Yoast SEO: by Team Yoast – 12.1
WP Accessibility: by Joe Dolson – 1.6.10
WP Product Feed Manager: by Michel Jongbloed – 1.15.0 – Not tested with the active version of WooCommerce
WP Sitemap Page: by Tony Archambeau – 1.6.2
Yoast SEO: WooCommerce: by Team Yoast – 11.6 – Not tested with the active version of WooCommerce

### Inactive Plugins (2) ###

AddressFinder: by Abletech – 1.3.1 – Not tested with the active version of WooCommerce
Akismet Anti-Spam: by Automattic – 4.1.2

### Must Use Plugins (1) ###

InfiniteWP - Client Loader: by Revmakx –

### Settings ###

API Enabled: –
Force SSL: –
Currency: AUD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: ✔

### WC Pages ###

Shop base: #51 - /products/
Cart: #4841 - /cart/
Checkout: #4842 - /checkout/
My account: #4843 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Pacific Springs
Version: 2.0.0
Author URL: https://www.get-business-online.com/about-2/
Child Theme: ✔
Parent Theme Name: Twenty Eleven
Parent Theme Version: 3.3
Parent Theme Author URL: https://en-au.wordpress.org/
WooCommerce Support: ✔

### Templates ###

Overrides: /home/gboldev/public_html/pacificsprings/wp-content/plugins/woo-checkout-optimization-cartimize/templates/woocommerce/checkout/form-coupon.php
/home/gboldev/public_html/pacificsprings/wp-content/plugins/woo-checkout-optimization-cartimize/templates/woocommerce/checkout/form-shipping.php version 3.0.9 is out of date. The core version is 3.6.0

Outdated Templates: ❌
					
					
						Learn how to update

### Action Scheduler ###

Complete: 5
Oldest: 2019-06-20 14:15:28 +1000
Newest: 2019-07-18 14:31:51 +1000

Pending: 4
Oldest: 2019-07-25 14:31:51 +1000
Newest: 2019-09-16 09:30:31 +1000

Canceled: 0
Oldest: –
Newest: –

In-progress: 0
Oldest: –
Newest: –

Failed: 0
Oldest: –
Newest: –

```
</details>

**Comment by juliaamosova** [comment]
Hi @galbaras,

While we appreciate you sharing your ideas with us, it doesn’t fit in with our current priorities for the project. At some point, we may revisit our priorities and look through the list of suggestions like this one to see if it warrants a second look. In the meantime, we are going to close this issue with the `votes needed` label and evaluate over time if this issue collects more feedback. 

Don’t be alarmed if you don’t see any activity on this issue for a while.

**Comment by woocommercebot** [comment]
Thanks for the suggestion @galbaras.

We'll keep an eye on the [popularity of this request](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md#feature-requests) :)

**Comment by galbaras** [comment]
This is not a suggestion. The code is not compliant with the schema definition (a bug).

**Comment by rangler2** [comment]
Is this still the current functionality? I wish it *would* report the Sale price...

In my case I have the opposite problem : We have set both a Sale £15 and Regular £20 price,  but only the Regular price appears in the schema markup: 
"offers":[{"@type":"Offer","price":"20.0000","priceValidUntil":"2023-12-31","priceSpecification":{"price":"20.0000","priceCurrency":"GBP","valueAddedTaxIncluded":"false"}

This causes a "Mismatched value (page crawl) [price]"  error in Google Merchant Centre because the product feed I had set up includes both Regular (£20) and Sale price (£15) attributes , but Google crawl is only seeing the regular price £20.

For now I had to remove the Sale price from google product feed so it only gets the regular price which matches what it sees on the website. Even though the website price is £5 lower when you click through. 

Am using the latest version of WooCommerce : 6.8.2
